packages/network-controller/src/create-network-client.ts

```diff
@@ -0,0 +1,188 @@
+import {
+  createAsyncMiddleware,
+  createScaffoldMiddleware,
+  JsonRpcEngine,
+  mergeMiddleware,
+  JsonRpcMiddleware,
+} from 'json-rpc-engine';
+import {
+  createBlockCacheMiddleware,
+  createBlockRefMiddleware,
+  createBlockRefRewriteMiddleware,
+  createBlockTrackerInspectorMiddleware,
+  createInflightCacheMiddleware,
+  createFetchMiddleware,
+  createRetryOnEmptyMiddleware,
+} from '@metamask/eth-json-rpc-middleware';
+import {
+  providerFromEngine,
+  providerFromMiddleware,
+  SafeEventEmitterProvider,
+} from '@metamask/eth-json-rpc-provider';
+import { createInfuraMiddleware } from '@metamask/eth-json-rpc-infura';
+import type { Hex } from '@metamask/utils/dist';
+import { PollingBlockTracker } from 'eth-block-tracker/dist';
+import { NetworksChainId } from '@metamask/controller-utils';
+
+const SECOND = 1000;
+
+export enum NetworkClientType {
+  Custom = 'custom',
+  Infura = 'infura',
+}
+
+type CustomNetworkConfiguration = {
+  chainId?: Hex;
```

Comment:
In the extension it appears that this is not optional. What's different here that makes this optional?